bundles/org.openhab.transform.scale/src/main/java/org/openhab/transform/scale/internal/ScaleTransformationService.java

```diff
@@ -129,7 +129,7 @@ private String formatResult(Map<Range, String> data, String source, final BigDec
 
     private String getScaleResult(Map<Range, String> data, String source, final BigDecimal value)
             throws TransformationException {
-        return data.entrySet().stream().filter(entry -> entry.getKey().contains(value)).findFirst()
+        return data.entrySet().stream().filter(entry -> entry.getKey() != null && entry.getKey().contains(value)).findFirst()
```

Comment:
Are you sure that the null value is the `entry.getKey())`? I'd thin it is actually the result of the `filter(...)` call and thus the `findFirst()` call causes the NPE.